renderdoc/driver/vulkan/vk_resources.cpp

```diff
@@ -3216,6 +3216,322 @@ VkImageAspectFlags FormatImageAspects(VkFormat fmt)
+
+          if(multiviewViewMaskTable)
+          {
+            multiviewViewMaskTable[remappedIndex] |= ci.pSubpasses[s].viewMask;
+          }
+        }
+        ++pResolveRunner;
+      }
+    }
+    if(pDepthStencilAttachment)
+    {
+      uint32_t index = pDepthStencilAttachment->attachment;
+
+      if(index < ci.attachmentCount)
+      {
+        VkAttachmentLoadOp depthStencilLoadOp = ci.pAttachments[index].loadOp;
+        VkAttachmentLoadOp stencilLoadOp = ci.pAttachments[index].stencilLoadOp;
+
+        uint32_t remappedIndex = indexRemapTable[index];
+        RDCASSERT(remappedIndex < arrayCount);
+
+        if(IsStencilFormat(ci.pAttachments[index].format))
+        {
+          // VK_KHR_separate_depth_stencil_layouts
+          VkAttachmentDescriptionStencilLayoutKHR *separateStencil =
+              (VkAttachmentDescriptionStencilLayoutKHR *)FindNextStruct(
+                  &ci.pAttachments[index],
+                  VK_STRUCTURE_TYPE_ATTACHMENT_DESCRIPTION_STENCIL_LAYOUT_KHR);
+
+          if(separateStencil)
+          {
+            loadOpTable[remappedIndex + 1] = stencilLoadOp;
+          }
+          else
+          {
+            // make depthstencil VK_ATTACHMENT_LOAD_OP_LOAD if either depth or stencil is
+            // VK_ATTACHMENT_LOAD_OP_LOAD
+            if(depthStencilLoadOp != VK_ATTACHMENT_LOAD_OP_LOAD)
+            {
+              depthStencilLoadOp = stencilLoadOp;
+            }
+          }
+        }
+
+        loadOpTable[remappedIndex] = depthStencilLoadOp;
+
+        if(multiviewViewMaskTable)
+        {
+          multiviewViewMaskTable[remappedIndex] |= ci.pSubpasses[s].viewMask;
+        }
+      }
+    }
+  }
+}
+
+RenderPassInfo::~RenderPassInfo()
+{
+  delete[] imageAttachments;
+  delete[] loadOpTable;
+  delete[] multiviewViewMaskTable;
+}
+
+FramebufferInfo::FramebufferInfo(const VkFramebufferCreateInfo &ci)
+{
+  // *2 in case we need separate barriers for depth and stencil, +1 for the terminating null
+  // attachment info
+  uint32_t arrayCount = ci.attachmentCount * 2 + 1;
+
+  imageAttachments = new AttachmentInfo[arrayCount];
+  RDCEraseMem(imageAttachments, arrayCount * sizeof(imageAttachments[0]));
+
+  width = ci.width;
+  height = ci.height;
+  layers = ci.layers;
+}
+FramebufferInfo::~FramebufferInfo()
+{
+  delete[] imageAttachments;
+}
+
+bool FramebufferInfo::AttachmentFullyReferenced(size_t attachmentIndex, const RenderPassInfo *rpi)
+{
+  VkResourceRecord *att = imageAttachments[attachmentIndex].record;
+  if(att->resInfo->imageInfo.extent.width != width || att->resInfo->imageInfo.extent.height != height)
+  {
+    return false;
+  }
+
+  if(rpi->multiviewViewMaskTable)
+  {
+    // check and make sure all views are referenced by the renderpass
+    uint32_t renderpass_viewmask = rpi->multiviewViewMaskTable[attachmentIndex];
+    uint32_t attachment_viewmask = (1 << att->viewRange.layerCount()) - 1;
+    return renderpass_viewmask == attachment_viewmask;
```

Comment:
I think this wrongly assumes that all multiview masks will be contiguous from 0. You could have a renderpass multiview mask that has bits 2 and 3 set, and an image view with 2 layers, and you'd compare 0xC against 0x3 here. Instead I think you want to do `Bits::CountOnes` on the `renderpass_viewmask` and compare that to the number of layers.